src/platforms/node/vaults/file.vault.ts

```diff
@@ -0,0 +1,100 @@
+import * as fs from 'fs'
+import { join } from 'path'
+import { Vault, VaultOptions } from '../../../vault'
+import { Level } from '../../../level'
+import { Format } from '../../../format'
+import { Log } from '../../../log'
+
+export interface ConsoleVaultOptions extends VaultOptions {
+  level: Level,
+  format: Format,
+  directory: string,
+  fileName: string,
+  useRotation: boolean,
+}
+
+export class FileVault extends Vault {
+
+  private readonly directory: string
+  private readonly fileName: string
+  private readonly useRotation: boolean
+  private fileStream: fs.WriteStream
+  private filePath: string
+
+  constructor(options: ConsoleVaultOptions) {
+    super(options)
+    this.directory = options.directory
+    this.fileName = options.fileName
+    this.useRotation = options.useRotation
+    this.filePath = this.getFilePath(this.directory, this.fileName, this.useRotation)
+    this.fileStream = this.setupFileStream(this.filePath)
+  }
+
+  private setupFileStream(filePath: string): fs.WriteStream {
+    return fs.createWriteStream(filePath, { flags: 'a' })
+  }
+
+  private getFilePath(directory: string, rawFileName: string, useRotation: boolean): string {
+    const dateSuffix = this.getDateSuffix(useRotation)
+    const fileName = `${ rawFileName }${ dateSuffix }.log`
+    return join(directory, fileName)
+  }
+
+  private getDateSuffix(useRotation: boolean): string {
+    return useRotation ? `-${ this.getFormattedDate(new Date()) }` : ''
+  }
```

Comment:
Would be nice if the ternaries true and false return values was split out onto their own lines. Something along these lines.

```
< Insert Some Boolean Condition >
? < Result in case of true > 
: < Result in case of false >
```
Makes it slightly easier to read :)

Reply:
I do not entirely agree with this. In my opinion, ternary expressions should be used when you have two small expressions that you need to choose between. If you have the need to break them up over multiple lines, then an if statement is the better option.

If you think that the contents of the ternary in this case is difficult to read, then i'll happily either extract the true-branch to a named construct or rewrite it as an if-statement.